marlowe-chain-sync/src/Language/Marlowe/Runtime/ChainSync/Database/PostgreSQL.hs

```diff
@@ -279,7 +588,7 @@ data AssetOut = AssetOut TxId TxIx SlotNo PolicyId AssetName Quantity
 -- For how to do this with postgresql-libpg. You can expose the underlying
 -- libpg connection from a `Session` using `withLibPQConnection`
 -- (see https://hackage.haskell.org/package/hasql-1.6.0.1/docs/Hasql-Connection.html#v:withLibPQConnection)
-commitBlocks :: CommitBlocks Session
+commitBlocks :: CommitBlocks Transaction
```

Comment:
Add a comment that explains why all database writes must be atomic (either via a transaction or by being a single statement) at the block level